src/sentry/integrations/msteams/webhook.py

```diff
@@ -143,6 +145,8 @@ def verify_signature(request):
 
 
 class MsTeamsWebhookMixin:
+    provider: ClassVar[str]
```

Comment:
so... mixins aren't typesafe -- I looked at this previously and I think this can just be converted into free functions (since nothing depends on actual `self` state and `provider` is a constant `'msteams'`)

Reply:
I also dislike the mixin pattern. I don't hate the idea of having these methods grouped semantically as a class, but it could easily be injected (or just instantiated on demand) rather than inherited.

Good point about the provider being constant as well.

Is that a change you'd want to see in this PR? Notwithstanding the above, I'm not sure exactly what you mean about the mixin not being typesafe. The `ClassVar` annotation is doing its job AFAICT.

Reply:
it is and it isn't -- it ensures internal type safety but isn't checked against the full mro when mixed in so it doesn't actually guarantee any runtime safety

ideally we'd kill this mixin while fixing this file

Reply:
Done: https://github.com/getsentry/sentry/pull/77586/commits/ee139e11efb49434936eb3e150c872af7f7920ee

I did want to preserve the logical grouping of related methods, so they're in a new module now.